RefuelService.App/Messaging/RefuelEventHandler.cs

```diff
@@ -0,0 +1,95 @@
+﻿using RefuelService.Core.Messaging;
+using RefuelService.Core.Models;
+using RefuelService.Core.Services;
+using System.Threading.Tasks;
+using Utf8Json;
+
+namespace RefuelService.App.Messaging
+{
+    public class RefuelEventHandler : IEventHandlerCallback
+    {
+        private readonly IRefuelService _refuelService;
+
+        public RefuelEventHandler(IRefuelService refuelService)
+        {
+            _refuelService = refuelService;
+        }
+
+        public async Task<bool> HandleEventAsync(EventTypes eventType, string message)
+        {
+            switch (eventType)
+            {
+                case EventTypes.ShipDocked:
+                    {
+                        return await HandleShipDocked(message);
+                    }
+                case EventTypes.ShipUndocked:
+                    {
+
+                        return await HandleShipUndocked(message);
+                    }
+                case EventTypes.ServiceRequested:
+                    {
+                        return await HandleServiceRequested(message);
+                    }
+                case EventTypes.Unknown:
+                    {
+                        return true;
+                    }
+            }
+
+            return true;
+        }
+
+        private async Task<bool> HandleShipUndocked(string message)
+        {
+
+            Ship receivedShip = JsonSerializer.Deserialize<Ship>(message);
+            Ship existingShip = await _refuelService.GetShipAsync(receivedShip.Id);
```

Comment:
Use C# default conventions:
For example: var receivedShip = JsonSerialier.Deserializer(message);